## 90_realpath.patch by Mike Hommey <glandium@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Use realpath() for loader_GetOriginalPathname, so that symlinks are
## DP: properly followed when determining where the current library lives.

Index: nss/mozilla/security/nss/lib/freebl/genload.c
===================================================================
--- nss.orig/mozilla/security/nss/lib/freebl/genload.c	2012-10-05 14:46:28.267227928 +0200
+++ nss/mozilla/security/nss/lib/freebl/genload.c	2012-10-05 14:46:44.047079362 +0200
@@ -27,6 +27,16 @@
  */
 static char* loader_GetOriginalPathname(const char* link)
 {
+#ifdef __GLIBC__
+    char* tmp = realpath(link, NULL);
+    char* resolved;
+    if (! tmp)
+    	return NULL;
+    resolved = PR_Malloc(strlen(tmp) + 1);
+    strcpy(resolved, tmp); /* This is necessary because PR_Free might not be using free() */
+    free(tmp);
+    return resolved;
+#else
     char* resolved = NULL;
     char* input = NULL;
     PRUint32 iterations = 0;
@@ -61,6 +71,7 @@
         input = NULL;
     }
     return input;
+#endif
 }
 #endif /* XP_UNIX */
 
